### PR TITLE
AISDK-233: RevAI NodeSDK should include premium diarization flag

### PR DESCRIPTION
**Pull request by eugenep-rev**
**Description**

As a RevAI customer, I want to use premium diarizarion. Currently, we do not expose it in our nodejs sdk, but GLG customer wants to test it, so w should add it.

- [x] Code
- [x] Tests

Source issue: https://revinc.atlassian.net/browse/AISDK-233
Docs PR: https://github.com/revdotcom/revai-docs/pull/296

### CODE REVIEW

File: src/models/async/DiarizationType.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable no-shadow */
+export enum DiarizationType {
+    STANDARD = 'standard',
+    PREMIUM = 'premium'
```
Comment (eugenep-rev) [inline src/models/async/DiarizationType.ts:4]: not sure what is the correct way to define it, followed `CaptionType` approach

**Comment by eugenep-rev** [comment]
I executed all commands which were recommended in readme: build, tests, build-examples. 

not sure if I need to execute integration tests though. @amikofalvy ? this note is a little unclear:
```
* Note that `integration-test` is currently configured to work with a certain account specified in our continuous integration build environment, as such for now you can check the automated continuous integration checks to pass the integration tests.
```

File: examples/async_transcribe_local_media_file.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ const token = require('./config/config.json').access_token;
                 'here'
             ]
         }],
+        diazation_type: 'premium', // Optional property which allows to define diarization type
```
Comment (amikofalvy) [inline examples/async_transcribe_local_media_file.js]: What is the correct spelling?

```suggestion
        diarization_type: 'premium', // Optional property which allows to define diarization type
```

File: examples/async_transcribe_media_from_url.js [hunk 3]
```diff
@@ -41,6 +41,7 @@ const token = require('./config/config.json').access_token;
                 'here'
             ]
         }],
+        diazation_type: 'premium', // Optional property which allows to define diarization type
```
Comment (kirillatrev) [inline examples/async_transcribe_media_from_url.js]: Spelling

**Review by kirillatrev — APPROVED** [review]
Other than spelling, LGTM

**Review by k-weng — APPROVED** [review]
LGTM

File: examples/async_transcribe_local_media_file.js (hunk 2 above)
Comment (eugenep-rev) [reply examples/async_transcribe_local_media_file.js]: @amikofalvy fixed

File: src/models/async/RevAiJobOptions.ts [hunk 4]
```diff
@@ -23,6 +24,7 @@ export interface RevAiJobOptions {
     skip_diarization?: boolean;
     skip_punctuation?: boolean;
     skip_postprocessing?: boolean;
+    diarization_type?: DiarizationType;
```
Comment (amikofalvy) [inline src/models/async/RevAiJobOptions.ts]: I think this should be a string type like `transcriber`

**Review by amikofalvy — CHANGES_REQUESTED** [review]
Need to change the types from enum to string

File: src/models/async/RevAiJobOptions.ts (hunk 4 above)
Comment (eugenep-rev) [reply src/models/async/RevAiJobOptions.ts]: can you please explain why it should be a string if we define an enum? not an expert in nodejs/typescript

Comment (amikofalvy) [reply src/models/async/RevAiJobOptions.ts]: It's a matter of responsibility. We can use the API to enforce which options are available without having to update the SDK.

Comment (amikofalvy) [reply src/models/async/RevAiJobOptions.ts]: Also, it follows the pattern of transcriber

**Comment by amikofalvy** [comment]
We also, will need to see a PR that updates the documentation.

**Comment by eugenep-rev** [comment]
> We also, will need to see a PR that updates the documentation.

https://github.com/revdotcom/revai-docs/pull/296

PR for the documentation

File: src/models/async/RevAiJobOptions.ts (hunk 4 above)
Comment (eugenep-rev) [reply src/models/async/RevAiJobOptions.ts]: fixed

File: src/models/async/RevAiJobOptions.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import { CustomerUrlData } from '../CustomerUrlData';
 import { CustomVocabulary } from '../CustomVocabulary';
+import { DiarizationType } from './DiarizationType';
```
Comment (amikofalvy) [inline src/models/async/RevAiJobOptions.ts]: Fix this error.

**Review by amikofalvy — COMMENTED** [review]
Fix the lint warning then :shipit:

**Review by amikofalvy — APPROVED** [review]

**Pull request merged**